### ISSUE TITLE
PH2117: False negative

### ISSUE THREAD

**Issue by ynsehoornenborg** [philips-software/roslyn-analyzers#610]
Th `AvoidUnnecessaryWhereAnalyzer` does not throw a Diagnostics on this statement:
``` cs
string[] result;
result.Select(x => x.Length).Where(l => l == 0).Any();
```
